homeassistant/components/shelly/utils.py

```diff
@@ -462,3 +462,17 @@ def is_rpc_wifi_stations_disabled(
         return False
 
     return True
+
+
+async def parse_host(input_host: str) -> tuple[str, int]:
+    """Parse host to verify port."""
+
+    if ":" in input_host:
+        splited: list = input_host.split(":")
+        host = splited[0]
+        port = int(splited[1])
```

Comment:
This will raise a `ValueError` if the user provides a host with `:` but no port number, e.g. `192.168.2.56:`.

Reply:
You want to guard against ? Current code if you put a ":" in the host raise an error as well;

Reply:
Done